File: app/src/main/java/com/github/pockethub/android/ui/item/news/PullRequestEventItem.java
package com.github.pockethub.android.ui.item.news;

import android.support.annotation.NonNull;
import android.text.TextUtils;
import android.view.View;

import com.github.pockethub.android.ui.StyledText;
import com.github.pockethub.android.ui.view.OcticonTextView;
import com.github.pockethub.android.util.AvatarLoader;
import com.meisolsson.githubsdk.model.GitHubEvent;
import com.meisolsson.githubsdk.model.PullRequest;
import com.meisolsson.githubsdk.model.payload.PullRequestPayload;

public class PullRequestEventItem extends NewsItem {

    public PullRequestEventItem(AvatarLoader avatarLoader, GitHubEvent dataItem) {
        super(avatarLoader, dataItem);
    }

    @Override
    public void bind(@NonNull NewsItem.ViewHolder viewHolder, int position) {
        super.bind(viewHolder, position);
        viewHolder.icon.setText(OcticonTextView.ICON_PULL_REQUEST);

        PullRequestPayload payload = (PullRequestPayload) getData().payload();

        StyledText main = new StyledText();
        boldActor(main, getData());

        PullRequestPayload.Action action = payload.action();
        if (PullRequestPayload.Action.Synchronized.equals(action)) {
            main.append("updated");
        }
        main.append(' ');
        main.append(action.name().toLowerCase());
        main.append(' ');
        main.bold("pull request " + payload.number());
        main.append(" on ");

        boldRepo(main, getData());

        viewHolder.event.setText(main);

        StyledText details = new StyledText();
        if (PullRequestPayload.Action.Opened.equals(action) ||
                PullRequestPayload.Action.Closed.equals(action)) {
            PullRequest request = payload.pullRequest();
            if (request != null) {
                String title = request.title();
                if (!TextUtils.isEmpty(title)) {
                    details.append(title);
                }
            }
        }

        if (TextUtils.isEmpty(details)) {
            viewHolder.details.setVisibility(View.GONE);
        } else {
            viewHolder.details.setText(details);
        }
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/news/PullRequestEventItem.kt
package com.github.pockethub.android.ui.item.news

import android.text.TextUtils
import android.view.View
import androidx.core.text.bold
import androidx.core.text.buildSpannedString
import com.github.pockethub.android.core.issue.IssueEventMatcher
import com.github.pockethub.android.ui.issue.IssuesViewActivity
import com.github.pockethub.android.ui.view.OcticonTextView
import com.github.pockethub.android.util.AvatarLoader
import com.github.pockethub.android.util.ConvertUtils
import com.github.pockethub.android.util.android.text.clickable
import com.meisolsson.githubsdk.model.GitHubEvent
import com.meisolsson.githubsdk.model.payload.PullRequestPayload
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.news_item.*

class PullRequestEventItem(
        avatarLoader: AvatarLoader,
        gitHubEvent: GitHubEvent
) : NewsItem(avatarLoader, gitHubEvent) {

    override fun bind(holder: ViewHolder, position: Int) {
        super.bind(holder, position)
        holder.tv_event_icon.text = OcticonTextView.ICON_PULL_REQUEST

        val payload = gitHubEvent.payload() as PullRequestPayload
        val action = payload.action()

        holder.tv_event.text = buildSpannedString {
            val context = holder.root.context
            boldActor(context, this, gitHubEvent)
            if (PullRequestPayload.Action.Synchronized == action) {
                append("updated")
            }
            append(" ${action?.name?.toLowerCase()} ")
            bold {
                clickable(onClick = {
                    val issue = IssueEventMatcher.getIssue(gitHubEvent)
                    val repository = ConvertUtils.eventRepoToRepo(gitHubEvent.repo())
                    context.startActivity(IssuesViewActivity.createIntent(issue, repository))
                }) {
                    append("pull request " + payload.number())
                }
            }
            append(" on ")
            boldRepo(context, this, gitHubEvent)
        }

        val details = buildSpannedString {
            if (PullRequestPayload.Action.Opened == action
                    || PullRequestPayload.Action.Closed == action
            ) {
                val request = payload.pullRequest()
                if (request != null) {
                    val title: String? = request.title()
                    if (!TextUtils.isEmpty(title)) {
                        append(title)
                    }
                }
            }
        }

        if (TextUtils.isEmpty(details)) {
            holder.tv_event_details.visibility = View.GONE
        } else {
            holder.tv_event_details.visibility = View.VISIBLE
            holder.tv_event_details.text = details
        }
    }
}
